src/DefaultContainer.php

```diff
@@ -57,7 +57,7 @@ public function setInstance(string $name, $service): self
      * @see ContainerInterface::has()
      *
      * @param string $name
-     * @return mixed
+     * @return bool
```

Comment:
I actually wonder how this behaves with `psr/container:^2` - is omitting the return type allowed at all?

Reply:
I tested the following, hopefully this is the setup you were wondering about:

`PHP 7.4.29 (cli)`
`psr/container": "^2.0"`

Class:

```php
<?php

declare(strict_types=1);

namespace App;

use Psr\Container\ContainerInterface;

/**
 * Undocumented class
 */
class Container implements ContainerInterface
{
    /**
     * Undocumented function
     */
    public function get(string $name)
    {
    }

    /**
     * Undocumented function
     */
    public function has(string $name)
    {
        return false;
    }
}
```

Script:
```php
<?php

use App\Container;

include __DIR__ . '/vendor/autoload.php';

$c = new Container();
```
Result:

`Fatal error: Declaration of App\Container::has(string $name) must be compatible with Psr\Container\ContainerInterface::has(string $id): bool`

Reply:
Aye, guessed as much: a patch to upgrade to `psr/container:^2` will be messy and breaking BC too :-\

Reply:
That's unfortunate.